### PR TITLE
Get to ginkgo v2

### PR DESCRIPTION
**Pull request by moleske**
Update - panics all gone, one failing test as described in this [comment](https://github.com/cloudfoundry/cli/pull/2378#discussion_r1327866480) (below here) that I need help resolving.  Everything else seems to be working.  I think future work would need to give thought if it is worth trying to backport this at all to v8 and/or v7

old original description details

> - panic occuring because of mismatch ginkgo versions due to ginkgo v1 being needed by code.cloudfoundry.org/cfnetworking-cli-api
> - code.cloudfoundry.org/cfnetworking-cli-api is archived and hasn't been maintained for quite awhile
> - the error shows as /Users/moleske/workspace/cli/util/clissh/clissh.test flag redefined: ginkgo.seed panic: /Users/moleske/workspace/cli/util/clissh/clissh.test flag redefined: ginkgo.seed
> 
> goroutine 1 [running]:
> flag.(*FlagSet).Var(0x14000116120, {0x1049bcdf8, 0x104c3e280}, {0x1400011d020, 0xb}, {0x104842d23, 0x2a})
> 	/opt/homebrew/Cellar/go/1.20.1/libexec/src/flag/flag.go:982 +0x2a4
> flag.(*FlagSet).Int64Var(...)
> 	/opt/homebrew/Cellar/go/1.20.1/libexec/src/flag/flag.go:769
> github.com/onsi/ginkgo/config.Flags(0x1049370c0?, {0x10482abdb?, 0x104843acb?}, 0x1)
> 	/Users/moleske/go/pkg/mod/github.com/onsi/ginkgo@v1.16.4/config/config.go:75 +0xe4
> github.com/onsi/ginkgo.init.0()
> 	/Users/moleske/go/pkg/mod/github.com/onsi/ginkgo@v1.16.4/ginkgo_dsl.go:53 +0x34
> 
> Ginkgo ran 20 suites in 10.659707541s
> Test Suite Failed
> make: *** [units-non-plugin] Error 1

cc @cloudfoundry/wg-app-runtime-interfaces-cli-approvers if you all have thoughts
Thank you for contributing to the CF CLI! Please read the following:

* Please make sure you have implemented changes in line with the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* We're not allowed to accept any PRs without a signed CLA, no matter how small.
If your contribution falls under a company CLA but your membership is not public, expect delays while we confirm.
* All new code requires tests to protect against regressions.
* Contributions must be made against the appropriate branch. See the [contributing guidelines](https://github.com/cloudfoundry/cli/blob/main/.github/CONTRIBUTING.md)
* Contributions must conform to our [style guide](https://github.com/cloudfoundry/cli/wiki/CLI-Product-Specific-Style-Guide). Please reach out to us if you have questions.

## Does this PR modify CLI v6, CLI v7, or CLI v8?
nah
Please see the contribution doc above or review [Architecture Guide](https://github.com/cloudfoundry/cli/wiki/Architecture-Guide).

## Description of the Change
get to ginkgo v2 hotness

We must be able to understand the design of your change from this description.
Keep in mind that the maintainer reviewing this PR may not be familiar with or
have worked with the code here recently, so please walk us through the concepts.

## Why Is This PR Valuable?

What benefits will be realized by the code change? What users would want this change? What user need is this change addressing? 
cause ginkgo v1 not supported no more

## Why Should This Be In Core?

Explain why this functionality should be in the cf CLI, as opposed to a plugin. 

## Applicable Issues

List any applicable GitHub Issues here

## How Urgent Is The Change?

Is the change urgent? If so, explain why it is time-sensitive.

## Other Relevant Parties

Who else is affected by the change?

### CODE REVIEW

**Comment by moleske** [comment]
if you all have thoughts on proper way to remove code.cloudfoundry.org/cfnetworking-cli-api as a dependency give a holler

**Comment by a-b** [comment]
@moleske this is a great initiative - V2 brings new tools to navigate flakes. @ccjaimes and @pivotalgeorge might be interested in this too

**Comment by moleske** [comment]
reading the commit history of the archive repo that we depend on (https://github.com/cloudfoundry-incubator/cfnetworking-cli-api/commits/master) that is causing problems here - it seems like that repo was created by ripping it out of the cli code initially.  Seems reasonable to me to just bring it back if we're not maintaining it in that archived repo no longer

**Comment by moleske** [comment]
apparently same thing happened with https://github.com/cloudfoundry/gofileutils/commits/master

File: api/cfnetworking/networking_connection.go [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+		}
+
+		hostnameError := x509.HostnameError{}
+		if errors.As(err, &hostnameError) {
+			return networkerror.SSLValidationHostnameError{
+				Message: hostnameError.Error(),
+			}
+		}
+
+		return networkerror.RequestError{Err: e}
```
Comment (moleske) [inline api/cfnetworking/networking_connection.go:81]: @ccjaimes - I need help with this chunk of code.  I tried copying the ideas from https://github.com/cloudfoundry/cli/pull/2415 as the failures were the same issue.  But for some reason one test is not getting the same hostname error as in that pr

**Review by moleske — COMMENTED** [review]
~~I very specifically would like feedback on https://github.com/cloudfoundry/cli/pull/2378/commits/80e4086eee041127be22238e72633f34d5d331ab and https://github.com/cloudfoundry/cli/pull/2378/commits/9e7aaec884ae040d75d66e752eed2d998b286f00~~

~~I unilaterally made a decision in this pr to copy the code from those archived repos and move them here.  Since they are archived they no longer get updates.  They also weren't getting updates for years.  It seems they were one extracted long ago for reasons, but not clear to me why they were extracted long ago.  I'd like feedback if this seems like a correct move since that code was stale in those repos and they used to be here~~

I didn't actually need to change these cause I misunderstood where the ginkgo seed error was occurring.  I've reverted them but that doesn't excuse the fact they aren't maintained

File: api/cfnetworking/networking_connection.go (hunk 1 above)
Comment (moleske) [reply api/cfnetworking/networking_connection.go]: resolving as I reverted (see review comment)

**Comment by a-b** [comment]
We have a bunch of green units by disabling parallel execution.

**Comment by moleske** [comment]
integration tests are failing cause they are using ginkgo v1, as that's what is currently in the main branch.  this branch/pr updates the integration tests to use ginkgo v2.  which will cause problems with v8/v7 branches when this is merged.  This may be acceptable if we immediately work on porting those branches after this pr is merged

this pr won't go green until the main branch is updated to use ginkgo v2 in the integration tests

**Comment by a-b** [comment]
Delivered via #2810

**Pull request closed without merging**